multimodal/src/autogluon/multimodal/matcher.py

```diff
@@ -1065,35 +1153,66 @@ def _predict(
 
         return outputs
 
-    def evaluate(
+    def _evaluate_ranking(
         self,
-        data: Union[pd.DataFrame, dict, list],
+        qr_relevance: Union[pd.DataFrame, dict, list],
+        query_data: Union[pd.DataFrame, dict, list],
+        response_data: Union[pd.DataFrame, dict, list],
         id_mappings: Optional[Dict[str, Dict]] = None,
-        metrics: Optional[Union[str, List[str]]] = None,
-        return_pred: Optional[bool] = False,
+        chunk_size: Optional[int] = 1024,
+        similarity_type: Optional[str] = "cosine",
+        top_k: Optional[int] = 100,
     ):
-        """
-        Evaluate model on a test dataset.
+        query_header = self._query[0] if self._query is not None else QUERY
+        query_data = data_to_df(data=query_data, header=query_header)
+        response_header = self._response[0] if self._response else RESPONSE
+        response_data = data_to_df(data=response_data, header=response_header)
 
-        Parameters
-        ----------
-        data
-            A dataframe, containing the same columns as the training data
-        id_mappings
-             Id-to-content mappings. The contents can be text, image, etc.
-             This is used when the dataframe contains the query/response indexes instead of their contents.
-        metrics
-            A list of metric names to report.
-            If None, we only return the score for the stored `_eval_metric_name`.
-        return_pred
-            Whether to return the prediction result of each row.
+        qr_relevance = data_to_df(data=qr_relevance)
 
-        Returns
-        -------
-        A dictionary with the metric names and their corresponding scores.
-        Optionally return a dataframe of prediction results.
-        """
+        rank_labels = {}
```

Comment:
We can optimize this with FAISS in later commits.

Reply:
Sure.